src/ansible_navigator/configuration_subsystem/navigator_post_processor.py

```diff
@@ -46,13 +48,39 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+@dataclass
+class NavigatorVolumeMount:
+    """Describes EE volume mounts."""
+
+    src: str
+    dest: str
+    label: Optional[str] = None
+
+    def exists(self) -> bool:
+        """Does the volmount source exist?"""
+        return Path(self.src).exists()
+
+    def to_string(self) -> str:
+        """Renders the volume mount in a way that (docker|podman) understands."""
+        out = f"{self.src}:{self.dest}"
+        if self.label is not None:
+            out += f":{self.label}"
+        return out
+
+
 PostProcessorReturn = Tuple[List[LogMessage], List[ExitMessage]]
 
 
 class NavigatorPostProcessor:
     # pylint:disable=too-many-public-methods
     """application post processor"""
 
+    def __init__(self):
+        # Volume mounts accumulated from post processing various config entries.
+        # These get processed towards the end, in the (delayed)
+        # execution_environment_volume_mounts() post-processor.
+        self.extra_volume_mounts: List[NavigatorVolumeMount] = []
```

Comment:
Pro tip: if we turn the comment into an _attribute docstring_, Sphinx will pick it up and render in the docs.
```suggestion
        self.extra_volume_mounts: List[NavigatorVolumeMount] = []
        """Volume mounts accumulated from post-processing various config entries.

        These get processed towards the end, in the (delayed)
        execution_environment_volume_mounts() post-processor.
        """
```

Reply:
@cidrblock (educational hint) FYI in case you didn't know about the existence of attr docstrings :point_up:

Reply:
Sure, but this is an implementation detail; I'm not sure why we'd want this in user-facing docs.

Reply:
@webknjaz this? https://www.python.org/dev/peps/pep-0224/

That's cool......... There are a few places that could be useful where there are a ton of attrs, I'm thinking about some of the runner stuff.

Reply:
@relrod I think @webknjaz is referring to the dev/api docs in this case, we don't expose them today but I think it would be cool if we did.

They look like crap now because I didn't add good doc strings last year in a lot of what I wrote.

Reply:
> Sure, but this is an implementation detail; I'm not sure why we'd want this in user-facing docs.

The current Sphinx is set up to build a number of documents directly from the Python files. They are marked as "private API". This is not considered user-facing but is de-facto present in the Sphinx docs deployment.

Reply:
@cidrblock

> @webknjaz this? [python.org/dev/peps/pep-0224](https://www.python.org/dev/peps/pep-0224/)
> 
> That's cool......... There are a few places that could be useful where there are a ton of attrs, I'm thinking about some of the runner stuff.

Interesting... I'm not sure if I saw this rejected PEP so these docstrings are probably not made available in runtime and a treated as no-op statements syntax-wise. But Sphinx is able to parse them out. Also, there's an alternative syntax that makes use of comments. So instead here's a different way of doing the same (using `#:`):
```suggestion
        #: Volume mounts accumulated from post processing various config entries.
        #: These get processed towards the end, in the (delayed)
        #: execution_environment_volume_mounts() post-processor.
        self.extra_volume_mounts: List[NavigatorVolumeMount] = []
```

Ref: https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-autoattribute

I personally use the "docstring" style normally but there's no strong reason not to use `#:` instead.